pkgs/development/go-packages/generic/default.nix

```diff
@@ -48,26 +48,58 @@ let
 
   removeExpr = refs: ''remove-references-to ${lib.concatMapStrings (ref: " -t ${ref}") refs}'';
 
+  fetchSrc = fetch:
+    if fetch.type == "git" then
+      fetchgit {
+        inherit (fetch) url rev sha256;
+      }
+    else if fetch.type == "hg" then
+      fetchhg {
+        inherit (fetch) url rev sha256;
+      }
+    else if fetch.type == "bzr" then
+      fetchbzr {
+        inherit (fetch) url rev sha256;
+      }
+    else if fetch.type == "FromGitHub" then
+      fetchFromGitHub {
+        inherit (fetch) owner repo rev sha256;
+      }
+    else abort "Unrecognized package fetch type: ${fetch.type}";
+
   dep2src = goDep:
-    {
+    let
+      otherMajorVersions = map (v: {
+        src = fetchsrc v.fetch;
+        inherit (v) majorVersion;
+      }) (goDep.otherMajorVersions or []);
+
+      main = fetchSrc goDep.fetch;
+
+      combined = stdenv.mkDerivation {
+        inherit (main) name;
+
+        unpackCmd = "cp -r $curSrc/* ./";
```

Comment:
Since you override the behavior of unpackFile here, would it not more readable to just inline the `cp` command to make it easier to read?